pallets/deeper-node/src/lib.rs

```diff
@@ -745,9 +751,20 @@ pub mod pallet {
 
     impl<T: Config> NodeInterface<T::AccountId> for Module<T> {
         fn im_offline(account_id: &T::AccountId) -> bool {
-            let block = Self::get_im_online(account_id);
+            let block = Self::get_im_online(account_id).unwrap_or(T::BlockNumber::default());
+            let current_block = <frame_system::Module<T>>::block_number();
+            current_block - block >= T::DayToBlocknum::get().into()
+        }
+
+        fn im_ever_online(account_id: &T::AccountId) -> bool {
+            Self::get_im_online(account_id) != None
+        }
+
+        fn get_days_offline(account_id: &T::AccountId) -> u32 {
+            let block = Self::get_im_online(account_id).unwrap_or(T::BlockNumber::default());
```

Comment:
Should we add a checkout condition : 
`if Self::im_ever_online（account_id）== false {
 return 0u32}`

Reply:
Yes, I thought about this. It really depends on how we gonna interpret this.
Current implementation means if a device is never online, we deem it as offline. There should be consistency between the `im_offline` and `get_days_offline` logic. If `im_offline` returns `true`, `get_days_offline` >= `1`, and vice versa. If `im_offline` returns false, `get_days_offline` = `0`, and vice versa.

My thinking is, if `im_ever_online` = `false`, it makes logical sense for `im_offline` = `true`

Please let me know if there is any reason that invalidates the logic